README.md

```diff
@@ -1 +1,364 @@
+2. Stop your Bot server with the command: `./gradlew stop`
+3. Wipe away all of your current changes with the command: `git checkout -- .`
+4. Check out the next checkpoint branch with the command: `git checkout checkpoint-1` (or whatever checkpoint is next, checkpoint names are labelled in section headers)
+5. Start your bot server with the command: `./gradlew start`
+
+
+## Implementing the registration procedure [checkpoint-1]
+
+
+First, update `StartupProcedure` to `@Inject` a `RegistrationBean` and call `joinRound()`. When you save the change, you should see print statements in your logs indicating `joinRound()` was invoked. Your code should now look something like this:
+
+```java
+@WebListener
+public class StartupProcedure implements ServletContextListener {
+
+    @Inject
+    RegistrationBean regBean;
+
+    @PostConstruct
+    public void joinFirstRound() {
+        regBean.joinRound();
+    }
+}
+```
+
+In order to register our AI Bot with the core services so it can play in games, we need to do two things:
+get the current party ID from the core services
+join the party queue so our bot waits its turn to play games, and can be notified when it's time to join a game
+
+
+We can get the current party ID by invoking `GET http://<game-service-ip-port>/party/describe`.  The response will be a JSON with the format:
+
+```
+{
+  "isSingleParty" : boolean,
+  "partyId" : String
+}
+```
+
+
+### Making REST calls with MicroProfile REST Client
+
+
+Go to the `GameServiceClient` interface and add an interface method that represents the endpoint we want to invoke like so:
+
+```java
+@GET    
+@Path("/party/describe")
+@Produces(MediaType.APPLICATION_JSON)
+public Map<String, Object> describe();
+```
+
+If you are familiar with JAX-RS, you will notice that this is the exact same signature and annotations that a JAX-RS method would have, but without the method body.
+Next, we need to declare the base URI of this REST client. We can do this using the MP REST Client annotations on the interface:
+
+```java
+@Path("/")
+@RegisterRestClient(baseUri = "http://localhost:8080", configKey = AIConfiguration.GAME_SERVICE_KEY)
+public interface GameServiceClient {
+```
+
+When we run the game-service locally, its URI is http://localhost:8080, so we will default to using that with the `baseUri` attribute. The `configKey` attribute allows us to override the `baseUri` using MP Config, which we will do later on in section 5.
+
+
+Now that we have our REST Client bean, we can inject it into the `RegistrationBean` and get the current round id. Update the `RegistrationBean` class to have the following code:
+
+```java
+// don't forget to add these 2 annotations!
+@Inject
+@RestClient
+GameServiceClient gameService;
+
+public void joinRound() {
+    System.out.println("Attempting to register with game service...");
+
+    // TODO: use MPRestClient to pull the valid partyID from the game service
```

Comment:
the TODO should probably be taken out since you are implementing it in the provided code block.